osu.Game/Rulesets/Difficulty/DifficultyCalculator.cs

```diff
@@ -32,10 +32,6 @@ protected virtual void ApplyMods(Mod[] mods)
             TimeRate = clock.Rate;
         }
 
-        protected virtual void PreprocessHitObjects()
-        {
-        }
-
-        public abstract double Calculate(Dictionary<string, double> categoryDifficulty = null);
+        public abstract double Calculate(Dictionary<string, object> categoryDifficulty = null);
```

Comment:
This could do with some documentation (what is the nullable dictionary parameter?)